examples/parallel_profiling.py

```diff
@@ -0,0 +1,164 @@
+            (toc - tic) / n_iter)
+
+
+def print_info(mem_used_avg, elapsed_time):
+    print('Memory used by cropping the signal in the parent'
+          ' function: {:.2f} MiB'.format(mem_used_avg))
+    print('Elasped processing time: {:.2f} s\n'.format(elapsed_time))
+
+
+###############################################################################
+# As explained in the introduction, we will create a large array
+# (i.e. `signal`) and take is average at different locations (i.e. `slices`)
+# using a specific window size (i.e. `window_size`). To speed-up the execution,
+# we under-sample the number of slices to evaluate.
+
+import numpy as np
+
+signal = np.random.random((10000000,))
+window_size = 500000
+slices = [slice(start, start + window_size)
+          for start in range(0, signal.size - window_size, 50000)]
+
+
+###############################################################################
+# In the first strategy, the function passed to :class:`joblib.Parallel` will
+# take the portion of the signal from which we need to compute the
+# sum. Therefore, the function passed to parallel only compute the sum of the
+# array passed as argument. Therefore, all arrays passed to the
+# `moving_average` function will always be different and it will not be able to
+# share memory between the workers.
+
+from joblib import Parallel, delayed
+
+
+def process_cropping(signal, window_size, slices):
+    def moving_average(signal):
+        return signal.sum()
+
+    Parallel(n_jobs=2, max_nbytes=0)(delayed(
+        moving_average)(signal[sl]) for sl in slices)
+
+
+mem_used, elapsed_time = time_memory_profiling(process_cropping, signal,
+                                               window_size, slices)
+print_info(mem_used, elapsed_time)
+
+
+###############################################################################
+# The second strategy differs from the former since the original large array
+# will be passed to the `moving_average` function and the slicing will occur
+# directly in this function. Therefore, the original array can be shared
+# between the different workers and will lead to a more efficient use of the
+# memory.
+
+
+def process_slicing(signal, window_size, slices):
+    def moving_average(signal, sl):
+        return signal[sl].sum()
+
+    Parallel(n_jobs=2, max_nbytes=0)(delayed(
+        moving_average)(signal, sl) for sl in slices)
+
+
+mem_used, elapsed_time = time_memory_profiling(process_slicing, signal,
+                                               window_size, slices)
+print_info(mem_used, elapsed_time)
+
+###############################################################################
+# As a conclusion, the amount of memory used is reduced by passing the large
+# array and sharing it between the workers, even if the coding pattern seems
+# less intuitive. However, when looking at the execution time, this strategy is
+# more time consuming. This due to the fact that internally, an hash of the
```

Comment:
Maybe you don't need to talk about a hash actually here, you just need to tell that the array is memmapped before sending it to child processes. This is to allow sharing the arrays in child processes instead of having a copy of the array in each child process.